rate-limiter/src/main/scala/it/pagopa/interop/commons/ratelimiter/akkahttp/RateLimiterDirective.scala

```diff
@@ -20,26 +17,23 @@ import it.pagopa.interop.commons.ratelimiter.model.Headers.{
 import it.pagopa.interop.commons.ratelimiter.model.{Headers, RateLimitStatus}
 import it.pagopa.interop.commons.utils.AkkaUtils._
 import it.pagopa.interop.commons.utils.ORGANIZATION_ID_CLAIM
+import it.pagopa.interop.commons.utils.errors.AkkaResponses.tooManyRequests
+import it.pagopa.interop.commons.utils.errors.{GenericComponentErrors, ServiceCode}
 
 import scala.concurrent.ExecutionContext
 import scala.util.{Failure, Success}
 
 object RateLimiterDirective {
 
-  def rateLimiterDirective[T](rateLimiter: RateLimiter, tooManyRequestsProblem: T)(
-    contexts: Seq[(String, String)]
-  )(implicit
+  def rateLimiterDirective(rateLimiter: RateLimiter)(contexts: Seq[(String, String)])(implicit
     ec: ExecutionContext,
-    toEntityMarshallerProblem: ToEntityMarshaller[T],
+    serviceCode: ServiceCode,
     logger: LoggerTakingImplicit[ContextFieldsToLog]
-  ): Directive1[Seq[(String, String)]] =
-    rateLimit(rateLimiter, tooManyRequestsProblem)(contexts).flatMap(addHeaders)
+  ): Directive1[Seq[(String, String)]] = rateLimit(rateLimiter)(contexts).flatMap(addHeaders)
 
-  private def rateLimit[T](rateLimiter: RateLimiter, tooManyRequestsProblem: T)(
-    contexts: Seq[(String, String)]
-  )(implicit
+  private def rateLimit(rateLimiter: RateLimiter)(contexts: Seq[(String, String)])(implicit
     ec: ExecutionContext,
-    toEntityMarshallerProblem: ToEntityMarshaller[T],
+    serviceCode: ServiceCode,
```

Comment:
Is it me or `serviceCode` is unused in both functions?

Reply:
it is implicitly used in `tooManyRequests`